ExecuteJobs.java

```diff
@@ -0,0 +1,80 @@
+import java.util.*;
+import java.io.*;
+
+public class ExecuteJobs {
+  public long start(String process) throws IOException {
+    // TODO some error handling
+    Process p = Runtime.getRuntime().exec(process);
+    long pid = p.pid();
+    return pid;
+  }
+
+  public int stop(long pid) throws IOException {
+    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    String s = pInput.readLine();
+    if(s == null) {
+      // pid not found
+      return 2;
+    }
+
+    Runtime.getRuntime().exec("kill " + pid);
+    p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
```

Comment:
can you factor out process output read into separate sub-function for easier reading?

Reply:
Yes.